File: 53 Maximum Subarray/Solution.java
public class Solution {
    public int maxSubArray(int[] nums) {
        /*只要返回最终值，不用子序列*/
        int currmax = nums[0];
        int summax = nums[0];
        for (int i = 1; i < nums.length; i++) {
            currmax = currmax + nums[i] > nums[i] ? currmax + nums[i] : nums[i];
            summax = currmax > summax ? currmax : summax;
        }
        return summax;
    }
}

File: 53 Maximum Subarray/Solution.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def maxSubArray(self, nums):
        currmax = nums[0]
        summax = nums[0]
        for num in nums[1:]:
            currmax = currmax + num if currmax + num > num else num
            summax = currmax if currmax > summax else summax
        return summax
